neo/core/group.py

```diff
@@ -0,0 +1,58 @@
+"""
+This module implements :class:`Group`, which represents a subset of the
+channels in an :class:`AnalogSignal` or :class:`IrregularlySampledSignal`.
+
+It replaces and extends the grouping function of the former :class:`ChannelIndex`
+and :class:`Unit`.
+"""
+
+from neo.core.container import Container
+
+
+
+class Group(Container):
+    """
+    Can contain any of the data objects, views, or other groups,
+    outside the hierarchy of the segment and block containers.
+    A common use is to link the :class:`SpikeTrain` objects within a :class:`Block`,
+    possibly across multiple Segments, that were emitted by the same neuron.
+
+    *Required attributes/properties*:
+        None
+
+    *Recommended attributes/properties*:
+        :objects: (Neo object) Objects with which to pre-populate the :class:`Group`
+        :name: (str) A label for the group.
+        :description: (str) Text description.
+        :file_origin: (str) Filesystem path or URL of the original data file.
+
+    Note: Any other additional arguments are assumed to be user-specific
+            metadata and stored in :attr:`annotations`.
+
+    *Container of*:
+        :class:`AnalogSignal`, :class:`IrregularlySampledSignal`, :class:`SpikeTrain`,
+        :class:`Event`, :class:`Epoch`, :class:`View`, :class:`Group
+    """
+    _data_child_objects = (
+        'AnalogSignal', 'IrregularlySampledSignal', 'SpikeTrain', 'Event', 'Epoch', 'View'
+    )
+    _container_child_objects = ('Segment', 'Group')
+    _single_parent_objects = ('Block',)
+
+    def __init__(self, *objects, name=None, description=None, file_origin=None, **annotations):
```

Comment:
Maybe, I am old school but I would expect ```objects``` an iterable and not ```*objects``` here.

A standard use case will be
```python
a_list = []
for i in myloop:
    a_list.append(an_obbect)   

# This is more intuitive a biginer
g = Group(a_list)

# if I understand correctly you propose this
g = Group(*a_list)
```

Reply:
I agree